src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -340,17 +339,8 @@ private ILocalService GetConfigurationPersister()
 
         private ILocalService GetConfigurationProvider()
         {
-            var solution = this.GetService<SVsSolution, IVsSolution>();
-            var connectedModeConfigPathProvider = new ConnectedModeSolutionBindingPathProvider(solution);
-            var legacyConfigPathProvider = new LegacySolutionBindingPathProvider(solution);
-
             var store = this.GetService<ICredentialStoreService>();
-            var credentialsLoader = new SolutionBindingCredentialsLoader(store);
-            var bindingFileLoader = new SolutionBindingFileLoader(Logger);
-
-            var solutionBindingDataReader = new SolutionBindingDataReader(bindingFileLoader, credentialsLoader);
-
-            return new ConfigurationProvider(legacyConfigPathProvider, connectedModeConfigPathProvider, solutionBindingDataReader);
+            return new ConfigurationProvider(this, Logger, store);
```

Comment:
The credential store is a thin wrapper that just requires a logger to be constructed.
Does it need to be passed in to the ConfigurationProvider, or could it be new-ed up in the internal constructor?

Also, the convention with parameter-ordering is to put the logger last.

Reply:
It could be new-ed up but then it won't be the same store instance as used in ConfigurationPersister. They both use ALM's `SecretStore`, that has a cache in the implementation, and a locking mechanism surrounding it. We don't do a lot of read-write operations so perhaps it's not a problem, but I think it's still best to use the same instance. I'll change the order of the parameters.